DiscordChatExporter.Core/Exporting/ChannelExporter.cs

```diff
@@ -27,45 +27,35 @@ public async ValueTask ExportChannelAsync(
             );
         }
 
+        // Build context
+        var context = new ExportContext(discord, request);
+        await context.PopulateChannelsAndRolesAsync(cancellationToken);
+
+        // Export messages
+        await using var messageExporter = new MessageExporter(context);
+        await messageExporter.EnsureFileIsCreated(cancellationToken);
+
         // Check if the channel is empty
         if (request.Channel.IsEmpty)
         {
-            throw new DiscordChatExporterException(
-                $"Channel '{request.Channel.Name}' "
-                    + $"of guild '{request.Guild.Name}' "
-                    + $"does not contain any messages."
-            );
-        }
-
-        // Check if the 'after' boundary is valid
-        if (request.After is not null && !request.Channel.MayHaveMessagesAfter(request.After.Value))
-        {
-            throw new DiscordChatExporterException(
+            throw new ChannelNotExportedException(
```

Comment:
Still sounds like `ChannelEmptyException` is an okay name, given that we're not throwing any other exception to discern between "fully empty" and "circumstantially empty"